imod_coupler/drivers/ribametamod/ribametamod.py

```diff
@@ -401,36 +427,37 @@ def exchange_rib2mod(self) -> None:
 
     def exchange_sprinkling_demand_msw2rib(self, delt: float) -> None:
         # flux demand from metaswap sprinkling to Ribasim (demand)
-        if self.enable_sprinkling_surface_water:
-            self.msw_sprinkling_demand_sec = (
-                self.msw.get_surfacewater_sprinking_demand_ptr() / days_to_seconds(delt)
-            )
+        self.msw_sprinkling_demand_sec = (
+            self.msw.get_surfacewater_sprinking_demand_ptr() / days_to_seconds(delt)
+        )
+        assert "sw_sprinkling" in self.mapping.msw2rib
+        mapped = self.mapping.msw2rib["sw_sprinkling"].dot(
+            self.msw_sprinkling_demand_sec
+        )
 
-            ribasim_sprinkling_demand_sec = self.mapping.msw2rib["sw_sprinkling"].dot(
-                self.msw_sprinkling_demand_sec
-            )[:]
-            self.ribasim_infiltration += np.where(
-                ribasim_sprinkling_demand_sec > 0, ribasim_sprinkling_demand_sec, 0
-            )
-            self.ribasim_drainage += np.where(
-                ribasim_sprinkling_demand_sec < 0, -ribasim_sprinkling_demand_sec, 0
-            )
+        self.ribasim_user_demand[:] = mapped[:, np.newaxis] * self.ribasim_user_demand0
+        pass
 
-    def exchange_sprinkling_flux_realised_msw2rib(
-        self, realised_fractions: NDArray[np.float64]
-    ) -> None:
-        # realised flux from Ribasim to metaswap
-        if self.enable_sprinkling_surface_water:
-            msw_sprinkling_realised = self.msw.get_surfacewater_sprinking_realised_ptr()
-            # map fractions back to the shape of MetaSWAP array
-            msw_sprfrac_realised = self.mapping.msw2rib["sw_sprinkling"].T.dot(
-                realised_fractions
-            )
-            # multiply fractions with demands
-            msw_sprinkling_realised[:] = (
-                (self.msw_sprinkling_demand_sec * days_to_seconds(self.delt_gw))
-                * msw_sprfrac_realised
-            )[:]
+    def exchange_sprinkling_flux_realised_msw2rib(self, delt: float) -> None:
+        msw_sprinkling_realised = self.msw.get_surfacewater_sprinking_realised_ptr()
+        realised_fractions: NDArray[np.float64] = np.full_like(
+            self.ribasim_user_realised, 0.0
+        )
+
+        nonzero = self.ribasim_user_realised > 0.0
+        realised_fractions[nonzero] = (
+            self.ribasim_user_realised[nonzero] / days_to_seconds(delt)
+            / self.ribasim_user_demand.sum(axis=1).flatten()[nonzero]
+        )
+        assert "sw_sprinkling" in self.mapping.msw2rib
+        msw_sprfrac_realised = (
+            realised_fractions * self.mapping.msw2rib["sw_sprinkling"]
+        )
+        msw_sprinkling_realised[:] = (
+            (self.msw_sprinkling_demand_sec * days_to_seconds(self.delt_sw))
+            * msw_sprfrac_realised
+        )[:] 
+        self.ribasim_user_realised[:] = 0.0     # reset cummulative for the next timestep
```

Comment:
Best to do this at the beginning of the exchange_sprinkling_demand_msw2rib method?